src/stream/src/task/barrier_manager.rs

```diff
@@ -214,8 +214,19 @@ impl LocalBarrierManager {
                 managed_state.collect(actor_id, barrier);
             }
         }
+    }
 
-        Ok(())
+    /// When a actor exit unexpectedly, it should report this event using this function, so meta
+    /// will notice actor's exit while collecting.
+    pub fn notify_exit(&mut self, actor_id: ActorId) {
```

Comment:
We will only call this if the actor exits on **failure**. How about naming it `notify_failure`?